flink-clients/src/main/java/org/apache/flink/client/CliFrontend.java

```diff
@@ -790,39 +767,95 @@ else if (result instanceof TriggerSavepointFailure) {
 	}
 
 	/**
-	 * Sends a {@link org.apache.flink.runtime.messages.JobManagerMessages.DisposeSavepoint}
-	 * message to the job manager.
+	 * Asks the JobManager to dispose a savepoint.
+	 *
+	 * <p>There are two options for this:
+	 * <ul>
+	 * <li>Either the job the savepoint belongs to is still running, in which
+	 * case the user code class loader of the job is used.</li>
+	 * <li>Or the job terminated, in which case the user JARs have to be
+	 * uploaded before disposing the savepoint.</li>
+	 * </ul>
 	 */
-	private int disposeSavepoint(SavepointOptions options, String savepointPath) {
-		try {
-			ActorGateway jobManager = getJobManagerGateway(options);
-			logAndSysout("Disposing savepoint '" + savepointPath + "'.");
-			Future<Object> response = jobManager.ask(new DisposeSavepoint(savepointPath), clientTimeout);
+	private int disposeSavepoint(SavepointOptions options) throws Throwable {
```

Comment:
Have we actually tested somewhere that the jars we're uploading to the JobManager allow us to dispose the given savepoint with user code?